### PR TITLE
core: Replace manual endianness swap with htobe64() call

### PR DESCRIPTION
**Pull request by luk1337**
https://github.com/weechat/weechat/pull/2021#discussion_r1716381589

### CODE REVIEW

**Comment by flashcode** [comment]
It seems you need to add `#include <endian.h>`, this could possibly fix compilation on macOS.

**Comment by luk1337** [comment]
@flashcode can you restart macos build? It failed for no apparent reason.

**Comment by flashcode** [comment]
Done.

File: src/core/core-crypto.c [hunk 1]
```diff
@@ -23,6 +23,10 @@
 #include "config.h"
 #endif
 
+#if defined(__APPLE__)
```
Comment (luk1337) [inline src/core/core-crypto.c:26]: should I move it above config.h include?

**Comment by flashcode** [comment]
Also from the man page:

```
Similar  functions are present on the BSDs, where the required header file is
<sys/endian.h> instead of <endian.h>.  Unfortunately, NetBSD, FreeBSD, and glibc haven't
followed the original OpenBSD naming convention for these functions, whereby the nn
component always appears at the end of the function name (thus, for example, in NetBSD,
FreeBSD, and glibc, the equivalent of OpenBSDs "betoh32" is "be32toh").
```

So it could fail on FreeBSD, I'm going to test.

Also I think including the header should be done as well, I'm pretty sure this can fail on some OS/distributions without it.

File: src/core/core-crypto.c (hunk 1 above)
Comment (flashcode) [reply src/core/core-crypto.c:26]: I think it could be moved below, after all includes.
No other choice than defining this on macOS?

**Comment by luk1337** [comment]
> Also from the man page:
> 
> ```
> Similar  functions are present on the BSDs, where the required header file is
> <sys/endian.h> instead of <endian.h>.  Unfortunately, NetBSD, FreeBSD, and glibc haven't
> followed the original OpenBSD naming convention for these functions, whereby the nn
> component always appears at the end of the function name (thus, for example, in NetBSD,
> FreeBSD, and glibc, the equivalent of OpenBSDs "betoh32" is "be32toh").
> ```
> 
> So it could fail on FreeBSD, I'm going to test.
> 
> Also I think including the header should be done as well, I'm pretty sure this can fail on some OS/distributions without it.

https://man.freebsd.org/cgi/man.cgi?query=byteorder&sektion=9 has htobe64(), so should be ok?

Also, sys/endian.h doesn't exist on macos ;/

File: src/core/core-crypto.c (hunk 1 above)
Comment (luk1337) [reply src/core/core-crypto.c]: did that:

```
#ifdef htonll
#define htobe64 htonll
#endif
```

**Comment by flashcode** [comment]
It seems complicated to use these functions while being portable…
But I'm sure it's doable, I'll then make tests on Linux, FreeBSD, GNU/Hurd, Cygwin, Windows WSL.

**Comment by flashcode** [comment]
It's OK on FreeBSD, TOTP are valid, tests are OK.

**Comment by flashcode** [comment]
Thanks for the quick fix, merged (branches master and 4.3).

**Pull request closed without merging**